demosplan/DemosPlanCoreBundle/ResourceTypes/ProcedureMapSettingResourceType.php

```diff
@@ -0,0 +1,291 @@
+            $settingName,
+            SettingsFilter::whereProcedureId($procedureSetting->getProcedure()->getId())->lock(),
+            false);
+        Assert::countBetween($settings, 0, 1);
+
+        return array_pop($settings);
+    }
+
+    protected function convertStartEndCoordinatesToFlatList(?array $coordinates): string
+    {
+        return null === $coordinates ? '' : implode(',', [
+            $coordinates['start']['latitude'],
+            $coordinates['start']['longitude'],
+            $coordinates['end']['latitude'],
+            $coordinates['end']['longitude']]);
+    }
+
+    protected function convertCoordinatesToFlatList(?array $coordinates): string
+    {
+        return null === $coordinates ? '' : implode(',', [
+            $coordinates['latitude'],
+            $coordinates['longitude']]);
+    }
+
+    protected function convertFlatListToCoordinates(string $rawCoordinateValues, $expectedCoordinatePair): ?array
+    {
+        if ('' === $rawCoordinateValues) {
+            return null;
+        }
+
+        $rawCoordinateValues = explode(',', $rawCoordinateValues);
+        $coordinateValues = [];
+
+        foreach ($rawCoordinateValues as $value) {
+            $coordinateValues[] = (float) $value;
+        }
+
+        Assert::count($coordinateValues, $expectedCoordinatePair);
+
+        if (2 === $expectedCoordinatePair) {
+            return [
+                'latitude'  => $coordinateValues[0],
+                'longitude' => $coordinateValues[1],
+            ];
+        } elseif (4 === $expectedCoordinatePair) {
+            return [
+                'start' => [
+                    'latitude'  => $coordinateValues[0],
+                    'longitude' => $coordinateValues[1],
+                ],
+                'end' => [
+                    'latitude'  => $coordinateValues[2],
+                    'longitude' => $coordinateValues[3],
+                ],
+            ];
+        } else {
+            throw new InvalidArgumentException('Expected exactly two or four coordinate values');
+        }
+    }
+
+    protected function convertCoordinatesToJson(?array $coordinates): ?string
+    {
+        return null === $coordinates ? null : json_encode($coordinates, JSON_THROW_ON_ERROR);
+    }
+
+    protected function convertJsonToCoordinates(string $rawCoordinateValues): ?array
+    {
+        return '' === $rawCoordinateValues ? null : json_decode($rawCoordinateValues, true, 512, JSON_THROW_ON_ERROR);
+    }
+
+    protected function getAvailablePublicScales(): array
+    {
+        return $this->convertToListOfInt(str_replace(['[', ']'], '', (string) $this->globalConfig->getMapPublicAvailableScales()));
+    }
+
+    protected function convertListOfIntToString(array $values): string
+    {
+        return implode(',', $values);
+    }
+
+    /**
+     * @param string|list<string> $values
+     *
+     * @return list<int>
+     */
+    protected function convertToListOfInt(string|array $values): ?array
```

Comment:
Can it return null or can it not?

Reply:
done here aa4daa1d1c78295a9fae8b7fe4249e27b8915480